### PR TITLE
Merge changes from Desktop version 3.0.7.

### PR DESCRIPTION
**Pull request by jsolman**
## Changes
* Merge under the hood changes from Desktop version 3.0.7
* Enable MainNet commit / claim / compound

### CODE REVIEW

File: src/components/dex/Trade.vue [hunk 1]
```diff
@@ -165,9 +165,12 @@ export default {
...
     isOutOfDate() {
-      return this.$store.state.latestVersion && this.$store.state.latestVersion.testExchangeScriptHash
-        && this.$store.state.latestVersion.testExchangeScriptHash.replace('0x', '')
-          !== this.$services.assets.DEX_SCRIPT_HASH;
+      if (!this.$store.state.latestVersion) {
+        return true;
+      }
+      const currentNetworkLatestDexScriptHash = this.$store.state.currentNetwork.net === 'MainNet' ?
+        this.$store.state.latestVersion.prodExchangeScriptHash : this.$store.state.latestVersion.testExchangeScriptHash;
+      return currentNetworkLatestDexScriptHash.replace('0x', '') !== this.$store.state.currentNetwork.dex_hash;
```
Comment (schaeferCode) [inline src/components/dex/Trade.vue:172]: This method is reused exactly a number of times it seems. Is there a way to extract this out so we aren't repeating code?

File: src/services/dex.js [hunk 2]
```diff
@@ -1030,37 +1030,72 @@ export default {
...
+                  neo.applyTxToAddressSystemAssetBalance(dexAddress, res.tx, true);
+
                   setTimeout(() => {
                     this.withdrawSystemAsset(assetId, quantity, res.tx.hash, res.utxoIndex)
                       .then((res) => {
                         if (res.success) {
+                          store.commit('setSystemWithdrawMergeState', { step: 4 });
+
+                          neo.monitorTransactionConfirmation(res.tx, true)
+                            .then(() => {
```
Comment (schaeferCode) [inline src/services/dex.js:1074]: Maybe you can flatten this nested 'then...catch'. Try `return neo.monitorTransactionConfirmation(res.tx, true)` on line 1073. Then you should be able to take away a tab on the 'then' on line 1074. I hope that makes sense...

File: src/services/dex.js [hunk 3]
```diff
@@ -1069,7 +1104,14 @@ export default {
         this.withdrawNEP5(assetId, quantity)
           .then((res) => {
             if (res.success) {
-              resolve(res.tx);
+              alerts.success('Withdraw Relayed.');
+              neo.monitorTransactionConfirmation(res.tx, true)
+                .then(() => {
+                  resolve(res.tx);
```
Comment (schaeferCode) [inline src/services/dex.js:1110]: I would suggest flattening this 'then' as well.

File: src/services/dex.js [hunk 4]
```diff
@@ -1101,6 +1143,19 @@ export default {
         gas: 0,
       };
 
+      const handleRetry = () => {
+        setTimeout(() => {
+          this.ignoreWithdrawInputs(config);
+          this.markWithdraw(assetId, quantity, tryCount + 1)
+            .then((res) => {
```
Comment (schaeferCode) [inline src/services/dex.js:1150]: Here too

File: src/services/dex.js [hunk 5]
```diff
@@ -1259,112 +1274,118 @@ export default {
...
+
     config.tx.inputs.forEach((input) => {
-      _.set(assetUTXOsToIgnore, input.prevHash, input.prevIndex);
+      _.set(assetUTXOsToIgnore, `${input.prevHash}-${input.prevIndex}-${currentNetwork.net}`, currentNetwork.bestBlock.index);
     });
   },
 
   calculateWithdrawInputsAndOutputs(config, assetId, quantity) {
-    return new Promise((resolve, reject) => {
+    return new Promise(async (resolve, reject) => {
```
Comment (schaeferCode) [inline src/services/dex.js:1285]: I think it'd be worth it to break this method up into multiple smaller methods. There's A LOT that's going on in here. It's pretty hard to read as it is.

Comment (schaeferCode) [reply src/services/dex.js:1285]: By 'this method' I mean, `calculateWithdrawInputsAndOutputs`.

File: src/services/dex.js [hunk 6]
```diff
@@ -1462,7 +1486,7 @@ export default {
   },
 
   withdrawSystemAsset(assetId, quantity, utxoTxHash, utxoIndex, tryCount = 1) {
-    return new Promise((resolve, reject) => {
+    return new Promise(async (resolve, reject) => {
```
Comment (schaeferCode) [inline src/services/dex.js:1489]: This is probably another candidate for breaking up. IMO.

File: src/services/dex.js [hunk 7]
```diff
@@ -1882,96 +1934,129 @@ export default {
               alerts.success('Commit relayed, waiting for confirmation...');
               neo.monitorTransactionConfirmation(res.tx, true)
                 .then(() => {
-                  resolve(res.tx);
-                  this.fetchCommitState(wallets.getCurrentWallet().address);
+                  setTimeout(async () => {
```
Comment (schaeferCode) [inline src/services/dex.js:1937]: What's the purpose of the setTimeout here?

**Review by schaeferCode — APPROVED** [review]
I still need to review NEO.js but it looks good so far. I just have a few suggestions.

File: src/components/dex/Trade.vue (hunk 1 above)
Comment (jsolman) [reply src/components/dex/Trade.vue:172]: sure; take it up in a seperate PR

File: src/services/dex.js (hunk 2 above)
Comment (jsolman) [reply src/services/dex.js:1074]: I agree; there is a lot that I had flattened in this method already, but it needs more. Can happen in a subsequent PR.

File: src/services/dex.js (hunk 3 above)
Comment (jsolman) [reply src/services/dex.js:1110]: ditto

File: src/services/dex.js (hunk 4 above)
Comment (jsolman) [reply src/services/dex.js:1150]: ditto

File: src/services/dex.js (hunk 5 above)
Comment (jsolman) [reply src/services/dex.js:1285]: Agree. It's tech debt that we will need to prioritize separately.

File: src/services/dex.js (hunk 7 above)
Comment (jsolman) [reply src/services/dex.js:1937]: Ensuring the commit state will be consistent before retrieving it.

File: src/services/dex.js (hunk 6 above)
Comment (jsolman) [reply src/services/dex.js:1489]: ditto

**Pull request merged**